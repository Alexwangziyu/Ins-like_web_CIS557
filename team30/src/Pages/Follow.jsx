import React, { useState, useEffect } from 'react';
import Stack from '@mui/material/Stack';
// import axios from 'axios';
import TopBar from '../component/TopBar';
import SearchBar from '../component/SearchBar';
import { getUsers } from '../api/mock_api';
import searchimg from './Component 2.png';
// import Userdata from '../MOCK_DATA.json';

// Top 100 films as rated by IMDb users. http://www.imdb.com/chart/top

function Follow() {
  // const [itemData, setItemData] = useState([]);
  const [udata, setUdata] = useState([]);
  // useEffect(() => {
  //   axios.get('https://634623d5745bd0dbd377354a.mockapi.io/api/user')
  //     .then((res) => res.data)
  //     .then((data) => setItemData(data));
  // }, []);
  useEffect(() => {
    getUsers().then((res) => setUdata(res));
  }, []);
  return (
    <>
      <TopBar />
      <Stack spacing={2} sx={{ display: 'flex', alignItems: 'center' }}>
        <span style={{ height: '10vh' }} />
        <img src={searchimg} alt="search" />
      </Stack>
      <Stack spacing={2} sx={{ display: 'flex', alignItems: 'center' }}>
        <SearchBar placeholder="Enter a user here" data={udata} />
      </Stack>

    </>
  );
}

export default Follow;
