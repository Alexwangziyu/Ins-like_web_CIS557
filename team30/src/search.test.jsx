import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import Follow from './Pages/Follow';
import mockBackendAxios from './mock-backend-data';

mockBackendAxios();

test('When user enters username it is displayed', async () => {
  render(
    <MemoryRouter>
      <Follow />
    </MemoryRouter>,
  );
  const element = screen.getByRole('textbox');
  await userEvent.type(element, 'username1');
  expect(element).toHaveValue('username1');
});

// eslint-disable-next-line no-undef
test('Follow matches snapshot', async () => {
  const component = await act(async () => renderer.create(
    <MemoryRouter>
      <Follow />
    </MemoryRouter>,
  ));

  const tree = component.toJSON();
  // eslint-disable-next-line no-undef
  expect(tree).toMatchSnapshot();
});
