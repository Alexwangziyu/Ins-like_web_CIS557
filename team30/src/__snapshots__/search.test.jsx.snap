// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Follow matches snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary bg-gradient shadow"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        <img
          alt=""
          height="40"
          src="logo.png"
          width="115"
        />
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              onClick={[Function]}
            >
              Activity
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              id="search"
              onClick={[Function]}
            >
              Search
            </a>
          </li>
          <li
            className="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              className="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdown"
              role="button"
            >
              Todo
            </a>
            <ul
              aria-labelledby="navbarDropdown"
              className="dropdown-menu"
            >
              <li>
                <a
                  className="dropdown-item"
                  onClick={[Function]}
                >
                  My Profile
                </a>
              </li>
              <li>
                <a
                  className="dropdown-item"
                  onClick={[Function]}
                >
                  New Post
                </a>
              </li>
              <li>
                <hr
                  className="dropdown-divider"
                />
              </li>
              <li>
                <a
                  className="dropdown-item"
                  onClick={[Function]}
                >
                  Log Out
                </a>
              </li>
            </ul>
          </li>
        </ul>
        <span
          className="navbar-text"
        >
          <div
            className="navbar-brand marginleft=100px"
          >
            Error
          </div>
        </span>
        <span
          className="navbar-brand"
        >
          <a
            className="navbar-brand"
            href="http://localhost:3000/profile/null"
          >
            <img
              alt="showcase"
              className="mb-0 rounded-circle border border-light border-2 display: inline"
              style={
                Object {
                  "borderRadius": "50%",
                  "height": "2.7rem",
                  "objectFit": "cover",
                  "width": "2.7rem",
                }
              }
            />
          </a>
        </span>
      </div>
    </div>
  </nav>,
  <div
    className="css-1846q8b-MuiStack-root"
  >
    <span
      style={
        Object {
          "height": "10vh",
        }
      }
    />
    <img
      alt="search"
      src="Component 2.png"
    />
  </div>,
  <div
    className="css-1846q8b-MuiStack-root"
  >
    <div
      className="search"
    >
      <div
        className="searchInputs"
      >
        <input
          id="searchBar"
          onChange={[Function]}
          placeholder="Enter a user here"
          type="text"
          value=""
        />
        <div
          className="searchIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
]
`;
